File: java/app/src/main/java/com/youknow/movie/ui/details/DetailsContract.java
package com.youknow.movie.ui.details;

import com.youknow.domain.model.Movie;

public interface DetailsContract {
    interface View {
        void showProgressBar(int visibility);
        void onMovieLoaded(Movie movie);
        void onError(int msgResId);
        void hideError();
    }

    interface Presenter {
        void getMovie(String movieId);
    }
}

File: kotlin-rx/app/src/main/java/com/youknow/movie/ui/details/DetailsContract.kt
package com.youknow.movie.ui.details

import com.youknow.domain.model.Movie

interface DetailsContract {
    interface View {
        fun showProgressBar(visibility: Int)
        fun onMovieLoaded(movie: Movie)
        fun onError(msgResId: Int)
        fun hideError()
    }

    interface Presenter {
        fun unsubscribe()
        fun getMovie(movieId: String)

    }
}